packages/eslint-plugin/src/rules/no-shadow.ts

```diff
@@ -517,6 +519,28 @@ export default util.createRule<Options, MessageIds>({
       );
     }
 
+    /**
+     * Get declared line and column of a variable.
+     * @param  variable The variable to get.
+     * @returns The declared line and column of the variable.
+     */
+    function getDeclaredLocation(
+      variable: TSESLint.Scope.Variable,
+    ): { global: true } | { global: false; line: number; column: number } {
+      const identifier = variable.identifiers[0];
+      if (identifier) {
+        return {
+          global: false,
+          line: identifier.loc.start.line,
+          column: identifier.loc.start.column + 1,
+        };
+      } else {
+        return {
+          global: true,
+        };
+      }
```

Comment:
Nitty nit: could be simplified to `return identifier ? { ... } : { ... };`.

Reply:
you really do love your ternaries 😄 
*personally* I prefer to only use ternaries for when I want an expression.
I find in this case where I can do everything with statements - using an if statement is clearer and cleaner.